pkg/util/block.go

```diff
@@ -17,31 +19,36 @@ import (
 )
 
 func GetDevNameFromBDF(bdf string) (string, error) {
-	ne, err := helperutil.NewExecutor(commontypes.ProcDirectory)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to create executor")
-	}
+	const sysBlock = "/sys/block"
 
-	cmdArgs := []string{"-n", "--nodeps", "--output", "NAME"}
-	output, err := ne.Execute(nil, "lsblk", cmdArgs, types.ExecuteTimeout)
+	entries, err := os.ReadDir(sysBlock)
 	if err != nil {
-		return "", errors.Wrap(err, "failed to list block devices")
+		return "", errors.Wrap(err, "failed to read /sys/block")
 	}
 
-	devices := strings.Fields(string(output))
-	for _, dev := range devices {
-		link, err := os.Readlink("/sys/block/" + dev)
+	re := regexp.MustCompile(`/` + regexp.QuoteMeta(bdf) + `(/|$)`)
+
+	for _, entry := range entries {
+		dev := entry.Name()
+		devPath := filepath.Join(sysBlock, dev, "device")
+
+		if _, err := os.Readlink(devPath); err != nil {
+			// Ignore devices without a device symlink (like virtual devices: loop, ramdisk)
+			continue
```

Comment:
Why not just rely on `filepath.EvalSymlinks`?
https://github.com/golang/go/blob/77fc27972a2333c1e3a43cfa406276775ac0c579/src/path/filepath/symlink.go#L104-L107